src/expandableCalendar/__test__/index.spec.ts

```diff
@@ -0,0 +1,294 @@
+      driver.render();
+
+      it('should be expanded if initialPosition prop = "open"', () => {
+        expect(driver.isCalendarExpanded()).toBe(true);
+      });
+    });
+
+    describe('hideKnob', () => {
+      const driver = new ExpandableCalendarDriver(testIdExpandable, TestCase({expandableCalendarProps: {hideKnob: true}}));
+
+      beforeEach(() => {
+        jest.useFakeTimers();
+        driver.render();
+      });
+
+      it('should hide Knob when hideKnob props = true', () => {
+        expect(driver.getKnob()).toBeNull();
+      });
+    });
+  });
+
+  describe('Init', () => {
+
+    beforeEach(() => {
+      driver.render();
+    });
+
+    it("should display today's date", () => {
+      // CalendarList
+      expect(driver.getCalendarList()).toBeDefined();
+      expect(getMonthTitle(driver.getCalendarList().props.data[50])).toBe(getMonthTitle(today));
+
+      // WeekCalendar
+      expect(driver.getWeekCalendar()).toBeDefined();
+      const weekCalendarProps = driver.getWeekCalendar().props;
+      const initialScrollIndex = weekCalendarProps.initialScrollIndex;
+      const currentWeek = weekCalendarProps.data[initialScrollIndex];
+      expect(currentWeek).toBe(dashedToday);
+
+      // container
+      expect(driver.isCalendarExpanded()).toBe(false);
+
+      // events
+      expect(onMonthChange).not.toHaveBeenCalled(); // not called anyways...
+    });
+  });
+
+  describe('Knob', () => {
+    beforeEach(() => {
+      driver.render();
+    });
+
+    it('should expand expandable header ', () => {
+      driver.toggleKnob();
+      jest.runAllTimers();
+
+      expect(driver.isCalendarExpanded()).toBe(true);
+    });
+
+    it('should day press close expandable header', () => {
+      driver.toggleKnob();
+      jest.runAllTimers();
+      driver.selectDay(dashedToday);
+      jest.runAllTimers();
+      expect(driver.isCalendarExpanded()).toBe(false);
+    });
+  });
+
+  describe('CalendarList updates', () => {
+    describe('Day Press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it('should day press update date', () => {
+        driver.selectDay(dashedToday);
+
+        // WeekCalendar
+        const weekCalendarProps = driver.getWeekCalendar().props;
+        const initialScrollIndex = weekCalendarProps.initialScrollIndex;
+        const currentWeek = weekCalendarProps.data[initialScrollIndex];
+        expect(currentWeek).toBe(dashedToday);
+
+        // events
+        expect(onDateChanged).toHaveBeenCalled();
+        // expect(onMonthChange).toHaveBeenCalled();
+      });
+    });
+
+    describe('arrow press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it.each([[Direction.LEFT],[Direction.RIGHT]])(`should call onDateChanged and onMonthChanged to next month first day when pressing the %s arrow`, (direction: Direction) => {
```

Comment:
You have a % sign in the string... 3 more in the file.

Reply:
%s is used in it.each to read the string from the param in the test name (like in java string formatting)

Reply:
oh, I didn't know that! nice!